sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -74,14 +87,24 @@ def test_spline_transformer_input_validation(params, err_msg):
         SplineTransformer(**params).fit(X)
 
 
-def test_spline_transformer_manual_knot_input():
-    """Test that array-like knot positions in SplineTransformer are accepted.
+@pytest.mark.parametrize("extrapolation", ["continue", "periodic"])
+def test_spline_transformer_manual_knot_input(extrapolation):
+    """
+    Test that array-like knot positions in SplineTransformer are accepted.
     """
     X = np.arange(20).reshape(10, 2)
-    knots = [[0.5, 1], [1.5, 2], [5, 10]]
-    st1 = SplineTransformer(degree=3, knots=knots).fit(X)
+    knots = [[0, 1], [1, 2], [5, 10], [6, 11]]
```

Comment:
Why the change of knot positions?

Reply:
See this discussion:
https://github.com/scikit-learn/scikit-learn/pull/19483/files/4aaf4414c89b04e9f70fe09cdf576ac5c7a9c79e#r584184962

scipy was throwing an error when using `construct_fast` and integer value knot positions. I've changed the knots positions to cover this case.

Reply:
But before, there were floats like `0.5` and `1.5`, now its integers only. On top, it should work with regardless of dtype of knots, as long as numeric.

Reply:
I was lazy and wanted to cover both cases (list and integer values) with the same test. I've added a separate test and reverted the knots (the GitHub UI is displaying this incorrectly).